holidays/countries/thailand.py

```diff
@@ -684,12 +713,21 @@ def _add_with_observed(dt, holiday_name) -> None:
         }
         # For years with exact date data.
         if year in raeknakhwan_dates:
-            _add_with_observed(
-                date(year, *raeknakhwan_dates[year]), raeknakhwan
+            _add_observed(
+                self._add_holiday(
+                    raeknakhwan,
+                    date(year, *raeknakhwan_dates[year]),
```

Comment:
```suggestion
                    *raeknakhwan_dates[year],
```

Reply:
Thanks for the suggestion, implemented :)